python/pyspark/sql/pandas/utils.py

```diff
@@ -94,3 +94,33 @@ def require_minimum_pyarrow_version() -> None:
             errorClass="ARROW_LEGACY_IPC_FORMAT",
             messageParameters={},
         )
+
+
+def require_minimum_numpy_version() -> None:
+    """Raise ImportError if minimum version of NumPy is not installed"""
```

Comment:
numpy is a dependency of pandas, so it is available in connect;
but not a mandatary one in classic.

It seems `kde` actually only need `np.linspace`, whose logic is not complex, so we may remove the direct dependency of `numpy` later,

Reply:
Good point! Created https://issues.apache.org/jira/browse/SPARK-50020 as a followup.